microprofile/WFLY-11529_subsystem_metrics.asciidoc

```diff
@@ -0,0 +1,158 @@
+
+[NOTE]
+====
+The `:read-attribute` operation is called internally with `include-undefined-metric` set to `true` (as explained in https://issues.jboss.org/browse/WFCORE-4190[WFCORE-4190])
+====
+
+=== Subsystem description
+
+This feature adds 2 attributes to the `microprofile-metrics-smallrye` subsystem resource:
+
+* `exposed-subsystems` - a list of strings corresponding the name of subsystems that exposes their metrics in the HTTP endpoints.
+** By default, this attribute is not defined (so there is no metrics from subsystems that are exposed). The special character `*` can be used to specify that __all__ subsystems will
+expose their metrics.
+* `prefix` - A string to prepend to WildFly metrics that are exposed by the HTTP endpoint `/metrics` with the Prometheus output format.
+** By default, this attribute is not defined. However standalone configuration that provides the subsystem will provide the `${wildfly.metrics.prefix:wildfly}` value
+   so that metric names will be prepended by `wildfly_` (and the user can overrides it by providing the `wildfly.metrics.prefix` system property).
+
+[NOTE]
+====
+* the `exposed-subsystems` was already present in the subsystem but it had no impact since the WildFly metrics from subsystems were disabled in https://issues.jboss.org/browse/WFLY-11399[WFLY-11399] after it was found out that the first implementation would not allow aggregation of these metrics.
+With this feature, the `exposed-subsystems` will be relevant again.
+====
+
+[WARNING]
+====
+Having a subsystem be mentioned in `exposed-subsystems` might not be sufficient to expose its metrics. Some subsystems (such as `undertow` and `messaging-activemq`) do not enable their metrics/statistics by default (as it has some memory/performance impact). These metrics provides a `statistics-enabled` attribute to enable their metrics. If this attribute value is `false`, their metrics will not be exposed even if there are listed in `exposed-subsystems`. The standalone profiles distributed with WildFly provides expressions to be able to enable such metrics when the server is started (as explained in https://issues.jboss.org/browse/WFLY-11471[WFLY-11471]).
+====
+
+==== Metric name and labels
+
+The name of a WildFly metric is based on the subsystem that provides it as well as the name of the attribute from the management model.
+The name can also be prepended with a `prefix` (specified on the `/subsystem=microprofile-metrics` resource).
+The name can also be appended with the `_total` suffix if it corresponds to a Counter metrics (as explained in https://issues.jboss.org/browse/WFCORE-4238[WFCORE-4238])
+Other information are stored using labels.
+
+For example Undertow exposes a metric attribute `request-count` (which is a counter) for every Servlet in an application deployment.
+This attribute will be exposed to Prometheus with the name `wildfly_undertow_request_count_total`.
```

Comment:
replace `wildfly_undertow_request_count_total` with `wildfly_undertow_request_count`

Drop trailing `_total`

```
# HELP wildfly_undertow_request_count The number of requests this listener has served
# TYPE wildfly_undertow_request_count gauge
```

Reply:
this is valid for multiple places in the document

Reply:
adding `_total` is the convention for counters in Prometheus. We have https://issues.jboss.org/browse/WFLY-11578 to be able to distinguish them from gauges (which is the default metric type used by WildFly)